src/startupscriptgenerator/src/node/scriptgenerator.go

```diff
@@ -47,57 +47,16 @@ const inspectParamVariableName = "ORYX_NODE_INSPECT_PARAM"
 
 // Checks if the application insights needs to be enabled for the current runtime
 func shouldApplicationInsightsBeConfigured() bool {
-	nodeAppInsightsKeyEnv := os.Getenv(consts.UserAppInsightsKeyEnv)
 	nodeAppInsightsEnabledEnv := os.Getenv(consts.UserAppInsightsEnableEnv)
 
 	// Check if the application insights environment variables are present
-	if nodeAppInsightsKeyEnv != "" && nodeAppInsightsEnabledEnv != "" && nodeAppInsightsEnabledEnv != "disabled" {
+	if nodeAppInsightsEnabledEnv != "" && nodeAppInsightsEnabledEnv != "disabled" {
```

Comment:
So here we are only checking for value from `ApplicationInsightsAgent_EXTENSION_VERSION` variable and not the connection string? I thought we would be checking only for that going forward?

Reply:
potential bug :)

Reply:
All of the AI related environment variables should be gracefully handled by the applicationinsights agent itself. If a particular env var is missing, the agent will fail to startup, log a reason, and proceed with running the rest of customer code. Having any env var checks here would just be an extra layer of redundancy and it would be up to you whether or not to keep/maintain these checks.

Users who have manually installed the applicationinsights package may be configuring it via a connection string environment variable and have not enabled attach via the Portal. This is when only 1 would exist and not the extension_version env var